test-framework/test-suites/integration/tests/fixtures/add_data.py

```diff
@@ -196,17 +196,31 @@ def add_host_with_net(host):
 	host.run("stack add host backend-0-0 appliance=backend rack=0 rank=0")
 	host.run("stack add host interface backend-0-0 interface=eth0 network=test ip=192.168.0.3")
 
-@pytest.fixture(params = (True, False))
+@pytest.fixture(params = (1, 2, 3, 4))
 def stack_load(request, host):
-	if request.param:
+	if request.param == 1:
 		def _load_using_exec(dump_file, **kwargs):
 			kwargs_string = " ".join(f"{key}={value}" for key, value in kwargs.items())
 			return host.run(f"stack load {dump_file} exec=True {kwargs_string}")
 
 		return _load_using_exec
 
-	def _load_using_bash(dump_file, **kwargs):
+	if request.param == 2:
+		def _load_using_exec_document(dump_file, **kwargs):
+			kwargs_string = " ".join(f"{key}={value}" for key, value in kwargs.items())
+			return host.run(f"stack load document={dump_file} exec=True {kwargs_string}")
+
+		return _load_using_exec_document
+
+	if request.param == 3:
+		def _load_using_bash(dump_file, **kwargs):
+			kwargs_string = " ".join(f"{key}={value}" for key, value in kwargs.items() if key != "exec")
+			return host.run(f"stack load {dump_file} {kwargs_string} | bash -x")
+
+		return _load_using_bash
+
+	def _load_using_bash_document(dump_file, **kwargs):
 		kwargs_string = " ".join(f"{key}={value}" for key, value in kwargs.items() if key != "exec")
-		return host.run(f"stack load {dump_file} {kwargs_string} | bash -x")
+		return host.run(f"stack load document={dump_file} {kwargs_string} | bash -x")
 
```

Comment:
Seems like this would have been cleaner as 4 fixtures, instead of whatever black magic you are doing. KISS

Reply:
There's no way, that I've found anyways, to parametrize which fixture a test uses. This is the alternative solution pytest provides.

https://docs.pytest.org/en/latest/fixture.html#fixture-parametrize

Reply:
Yah, I'm not a super big fan of parameterize either. It is okay, and even recommended by most stuff I've read, to repeat yourself in test code, if it makes the logic more straightforward. This stuff is so complicated, when it fails it won't be obvious to a future dev why.

Reply:
@caladd I took a shot at simplifying the test suite somewhat. I left the fixture since it doesn't seem that complicated, and we will want every load integration test to run through all 4 essentially equivalent methods of loading a dump.json.